api/swagger.yaml

```diff
@@ -8490,6 +8490,19 @@ paths:
           in: "query"
           description: "The tag to associate with the image on the registry."
           type: "string"
+        - name: "platform"
+          in: "query"
+          description: |
+            Platform in the format `os[/arch[/variant]]` used to select
+            a platform-specific manifest if the image is an index/manifest list.
+
+            If the option is not set, then no specific manifest is select and the
+            whole image index is pushed.
+
+            When specified, the daemon checks if the image has any manifest that
+            matches the requested platform and pushes it instead of the whole index.
+            If there's no matching manifest it returns a `404` status.
+          type: "string"
```

Comment:
As I've noted in my other comment, IMO we should spend a bit more time thinking about the UX we want from `docker push` WRT platforms before we implement this in the API ("no is temporary, yes is forever"), but I seem to recall being in a minority on that in the maintainers call where we discussed this, so I'm not going personally to block this if other maintainers don't agree. :bow: